dts/bindings/regulator/regulator-pmic.yaml

```diff
@@ -125,3 +125,23 @@ child-binding:
           description: |
             Mask to apply to the current-levels value before writing it to the
             ilim-reg value to set the current limit
+
+  child-binding:
+    description: |
+      Sub-nodes for the regulator state in different Zephyr Power management states.
+
+    properties:
+      regulator-on:
+        type: boolean
+        required: false
+        description: regulator should be on.
+
+      regulator-off:
+        type: boolean
+        required: false
+        description: regulator should be off.
+
+      regulator-microvolt:
+        type: int
+        required: false
+        description: the default voltage which regulator would be set in.
```

Comment:
please, let's stick to Linux standard properties (e.g. `regulator-boot-on`). See related work here https://github.com/zephyrproject-rtos/zephyr/pull/52326

Reply:
These are similar to the Linux properties, however I dropped the `suspend` as these properties apply to the Zephyr `PM_STATE_ACTIVE` state as well.

Reply:
@gmarull, sounds ok to you?

Reply:
we should be following this: https://elixir.bootlin.com/linux/latest/source/Documentation/devicetree/bindings/regulator/regulator.yaml#L286. This is something that the PM subsystem should actually facilitate, but in absence of proper infrastructure, I guess we'll need to wait. However, nothing prevents us to at least follow the same bindings schema.

Reply:
@gmarull. I have rebased and updated to use the latest changes made to the regulator driver. I have also changed the property names to match Linux

Reply:
@gmarull I have rebased and updated the code base to use the latest changes made to the regulator API. Based on the new design, I have dropped the new properties added. Kindly take a look and let me know if you see any other issues.